### DISCUSSION TITLE
Considering compatibility with WireGuard in special network

### DISCUSSION THREAD

**Discussion by lekeeith** [rustdesk/rustdesk#6611] [General]
I have already deployed WireGuard, and MSTSC (Microsoft Remote Desktop Connection) is working. The subnet for my WireGuard is 10.8.0.1. Both WireGuard and RustDesk are deployed within Docker. The host machine's IP is 192.168.2.1. WireGuard is using a bridge network, while RustDesk is using a host network. Now, the issue is that when I set RustDesk's desktop network to 192.168.2.1 while having WireGuard enabled on computers in the 192.168.2.0/24 subnet, RustDesk cannot find the server. However, it works when WireGuard is disabled. On computers outside the 192.168.2.0/24 subnet with WireGuard enabled, I can access the content within the 192.168.2.0/24 subnet, but RustDesk cannot find the server. If I disable WireGuard, I cannot access the 192.168.2.0/24 subnet at all.

I insist on using WireGuard because it allows me to only open one port, which is more secure. Additionally, besides setting RustDesk's relay service to 192.168.2.1, I have also tried setting it to 10.8.0.1 and 172.20.0.2 (my WireGuard Docker IP), but none of these configurations were successful.

**Comment by xlionjuan** [comment]
You don't need to host RustDesk Server if you can already connect each other though VPN
